ngx_http_sxg_filter_module.c

```diff
@@ -537,15 +556,20 @@ static ngx_int_t ngx_http_sxg_header_filter(ngx_http_request_t* req) {
     if (ctx == NULL) {
       return NGX_ERROR;
     }
+
+    if (pthread_mutex_init(&ctx->lock, NULL) != 0) {
+      return NGX_ERROR;
+    }
     ctx->response = sxg_empty_raw_response();
     ctx->main_resource_loaded = false;
     ngx_http_set_ctx(req, ctx, ngx_http_sxg_filter_module);
   }
-
   if (req->parent != NULL) {
     return ngx_http_next_header_filter(req);
   }
 
+  pthread_mutex_lock(&ctx->lock);
```

Comment:
The above code only initializes `ctx->lock` when `ctx` != `NULL`, yet here calls `pthread_mutex_lock` unconditionally.

Reply:
Yes, it's fine.
`ctx` is uninitialized on first client request. But this function is called on subresource handling context or others (in honest, I don't know every pattern when this function is called). So it is important guarding `ctx` structure for concurrent access.

Reply:
Sorry I misread this code.